### ISSUE TITLE
Find dataset for probing

### ISSUE THREAD

**Issue by stitsyuk** [leobumjin/probe_lm#5]

**Comment by stitsyuk** [comment]
Possible datasets for probing task:

1. **GLUE (MNLI section)** - Natural Language Inference task. 
**Labels:** contradiction, neutral, entailment
**Example:** _Premise: How do you know? All this is their information again. Hypothesis: This information belongs to them._
**Answer:** _entailment_
**Dataset:** https://huggingface.co/datasets/glue/viewer/mnli/train

2. **SNLI (Stanford Natural Language Inference)** - same design as GLUE MNLI
**Labels:** contradiction, neutral, entailment
**Example:** _Premise: Two blond women are hugging one another. Hypothesis: The women are sleeping._
**Answer:** _contradiction_
**Dataset:** https://huggingface.co/datasets/snli

3. **HANS (Heuristic Analysis for NLI Systems)** - NLI evaluation dataset that tests specific hypotheses about invalid heuristics that NLI models are likely to learn. This is an interesting dataset to do probing since we can compare it to the implementation from the paper where the dataset was introduced (https://arxiv.org/pdf/1902.01007.pdf). They fine-tuned BERT for this task and we can compare if probing internal representations can have better performance for handling invalid heuristics.
**Labels:** entailment, non-entailment
**Example:** _The doctors supported the scientist. The scientist supported the doctors._
**Answer:** _non-entailment_
**Drawbacks:** Since the task was designed to demonstrate that models are likely to follow shallow heuristics and shortcuts, probing can also struggle and have poor results.
**Dataset:** https://github.com/tommccoy1/hans

4. **IMDb Movie Reviews** - binary sentiment analysis dataset consisting of 50,000 reviews from the Internet Movie Database (IMDb) labeled as positive or negative. In my opinion, probing can capture positive and negative concepts well from reviews because of important words for the classes (e.g. "bad, awful, boring, sad" for negative, "good, wonderful, interesting, incredible" for positive).
**Labels:** positive, negative
**Example:** _Ned aKelly is such an important story to Australians but this movie is awful. It's an Australian story yet it seems like it was set in America. Also Ned was an Australian yet he has an Irish accent...it is the worst film I have seen in a long time._
**Answer:** _negative_
**Drawbacks:** Some reviews are quite long, about 500 words. If our model can handle long input representations, than this dataset can be good for probing.
**Dataset:** https://huggingface.co/datasets/imdb

5. **Hate Speech Offensive** - dataset for hate speech and offensive language detection on tweets. I think that probing can show good results on this data since offensive speech consists "bad words", which are rare in common speech and most datasets. Therefore, this "bad concepts" will be captured well in internal representations.
**Labels:** offensive language, neither, hate speech
**Example:** _I spend my money how i want bitch its my business_
**Answer:** _offensive language_
**Drawbacks:** Since the dataset is about hate speech, every example contains swear words, and it is always complicated to write a paper with bad words in experiments or in examples.
**Dataset:** https://huggingface.co/datasets/hate_speech_offensive

6. **GoEmotions** - corpus of carefully curated comments extracted from Reddit, with human annotations to 27 emotion categories or neutral. It is interesting to check if it is possible to probe emotion concepts.
**Labels:** admiration, amusement, anger, annoyance, approval, caring, confusion, curiosity, desire, disappointment, disapproval, disgust, embarrassment, excitement, fear, gratitude, grief, joy, love, nervousness, optimism, pride, realization, relief, remorse, sadness, surprise, neutral
**Example:** _Man I love reddit._
**Answer:** _love_
**Drawbacks:** Many labels (28 emotions) can be hard to train using internal representations.
**Dataset:** https://huggingface.co/datasets/go_emotions?row=17

7. **Tweet sentiment extraction** - dataset of tweets with sentiment classes for every sentence. Probing can capture positive and negative concepts well because of important words.
**Labels:** positive, neutral, negative
**Example:** _I really really like the song Love Story by Taylor Swift_
**Answer:** _positive_
**Dataset:** https://huggingface.co/datasets/mteb/tweet_sentiment_extraction

8. **BoolQ** - question answering dataset for yes/no questions. This dataset is close to true/false SAPLMA dataset, but has a little different design: it has 2 sentences: question and passage, and then the true/false label.
**Labels:** true, false
**Example:** _Question: did abraham lincoln write the letter in saving private ryan? Passage: In the 1998 war film Saving Private Ryan, General George Marshall (played by Harve Presnell) reads the Bixby letter to his officers before giving the order to find and send home Private James Francis Ryan after Ryan's three brothers died in battle._
**Answer:** _true_
**Drawbacks:** Some passages are long. Also, since we already have many true/false dataset from SAPLMA, maybe it is better to introduce more experiments on different tasks for more variety.
**Dataset:** https://huggingface.co/datasets/boolq

**Comment by leobumjin** [comment]
* HANS
* IMDB 

I will find knowledge neuron papers for these works.
